src/Board.cpp

```diff
@@ -64,7 +150,59 @@ Lines Board::getCluesRows() const
     return cluesRows_;
 }
 
+bool Board::isSolved()
+{
+    for(auto line : board_)
+        if (std::any_of(line.cbegin(),
+                        line.cend(),
+                        [](const auto elem)
+                        {
+                            return elem == BoardFields::UNKNOWN;
+                        }))
+            return false;
+
+    for(int row = 0; row < sizeRows_; row++)
+        if(not verifyLine(getRow(row), cluesRows_[row]))
+            return false;
+
+    for(int col = 0; col < sizeCols_; col++)
+        if(not verifyLine(getCol(col), cluesCols_[col]))
+            return false;
+
+    return true;
+}
+
 void Board::display() const
 {
     std::cout << DisplayBoard::display(*this);
 }
+
+bool verifyLine(BLine line, Line clues)
+{
+    Line result;
+    bool continous = false;
+    unsigned int cnt = 0;
+
+    for(auto elem : line)
```

Comment:
extract to a function named countContinuousBlackFields

Reply:
is this necessery? it will only by one line shorter and verify function will only contain: 
return clues == countCont...;

Reply:
done